authz/grpc_authz_server_interceptors.go

```diff
@@ -153,7 +154,7 @@ func (i *FileWatcherInterceptor) updateInternalInterceptor() error {
 	if err != nil {
 		return err
 	}
-	atomic.StorePointer(&i.internalInterceptor, unsafe.Pointer(interceptor))
+	i.prevInternalInterceptor = atomic.SwapPointer(&i.internalInterceptor, unsafe.Pointer(interceptor))
```

Comment:
Who/where is `prevInternalInterceptor` used? Is it simply to have a reference to the old `StaticInterceptor`, so that it is not cleaned up by the garbage collector?

Reply:
Yes, that's right. I have included the bug which contains stack trace for the crash.
https://b.corp.google.com/issues/243670542#comment4

Reply:
@easwars If you look at the stacktrace https://screenshot.googleplex.com/4ebtpWCURfbcWpM.png
When making authorization decision, it is possible we load the internalInterceptor correctly, but before we can invoke StreamInterceptor, we have replaced the old interceptor with new values and the old interceptor could get garbage collected causing "invalid memory address or nil pointer dereference"

I have tried to reproduce the error (https://github.com/grpc/grpc-go/pull/5626#issuecomment-1355845192), but no luck so far. Please let me know if you have any insights. Since the switch team saw a crash, there could be a potential problem in code. Since it hasn't reappeared (or we may not be seeing it due to limitations on switch side to detect the cause of crashes), it makes it more difficult to debug.